api/v1alpha1/tls_types.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+type TLSSettings struct {
+	// Version details the minimum/maximum TLS protocol verison that
+	// should be supported by this listener.
+	// +optional
+	Version *TLSVersions `json:"version,omitempty"`
```

Comment:
do we need to setup a default value for this field in EG?

Reply:
Today EG simply doesn't specify any default, leaving the default up to Envoy itself. Maybe keep that behaviour, so that there will be backwards compatibility with previous EG versions?

Reply:
The default min version is `TLSv1_2`
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto
`Using TLS protocol versions below TLSv1_2 has serious security considerations and risks.`
what is the reason to use  version below TLS 1.2 ?

Reply:
In some cases there are legacy clients that don't support TLS 1.2 but require an older version.

Reply:
I would vote for being explicit here. Setting the default to "whatever Envoy wants" leaves behavior interesting undefined as the Envoy version changes.

As @arkodg notes, Envoy's current default minimum version is TLS 1.2, and there are some _very_ good reasons not to go below that. I'd vote to define Envoy Gateway's minimum as TLS 1.2, while still allowing users to select 1.0 or 1.1 if they need to.

(Allowing the _maximum_ to float with Envoy is OK with me, FTR.)

Reply:
(I'd also vote to call this stanza `versions` plural rather than `version` singular.)

Reply:
> I would vote for being explicit here. Setting the default to "whatever Envoy wants" leaves behavior interesting undefined as the Envoy version changes.

"Whatever Envoy wants" is the current default, and is also what would happen if there is no `ClientTrafficPolicy` attached at all. 

The `TLS_Auto` constant maps all the way to the Envoy Proxy's enum, which is defined as `⁣Envoy will choose the optimal TLS version.` Setting the minimum in EG to something other than `TLS_Auto` means that if Envoy Proxy fixes its default, maybe for solving some future CVE, then Envoy Gateway doesn't automatically benefit from the fix in Envoy Proxy.

> (I'd also vote to call this stanza versions plural rather than version singular.)

I'm thinking to maybe get rid of the `versions` intermediate level, and align more closely with how Envoy Proxy models it. I think this would make it clearer.

WDYT?

Reply:
yeah, we could rm `versions` in favor of `minVersion` & `maxVersion` , thoughts @kflynn ?

Reply:
`minVersion` and `maxVersion` work for me.

I would definitely prefer having Envoy Gateway dictate a particular minimum version for predictability, but I don't feel all that strongly there.

Reply:
Modified the version specification to remove the intermediate level.

I also set the default minimal version default to be 1.2 and the default maximal version to be 1.3. This is actually exactly the same as it was before, since this is Envoy Proxy's default. I agree that it makes sense to document it more clearly.